server/routes/settings.js

```diff
@@ -48,23 +58,28 @@ router.put("/:email/platform/:platform", jwtVerify, (req, res, next) => {
   });
 });
 
-router.get("/:email", jwtVerify, async (req, res, next) => {
-  SettingsModel.findOne({ email: req.params.email }, async (err, settings) => {
+
+// Gets users settings
+router.get("/:email", jwtVerify, (req, res, next) => {
+  SettingsModel.findOne({ email: req.params.email }).populate('companies').exec((err, settings) => {
     if (settings) {
+      let companyNames = [];
+      settings.companies.forEach(company => companyNames.push(company.name));
       const interface = new Interface();
-      const mentions = await interface.getNewestMentions(settings.companies);
+      const mentions = await interface.getNewestMentions(companyNames);
```

Comment:
can you also make it an async route since it is using await?
and can you pass the companyNames to the frontend too